test/boost/row_cache_test.cc

```diff
@@ -2531,6 +2531,7 @@ SEASTAR_TEST_CASE(test_exception_safety_of_update_from_memtable) {
             snap->fill_buffer().get();
 
             cache.update(row_cache::external_updater([&] {
+                memory::scoped_critical_alloc_section dfg;
                 auto mt2 = make_memtable(cache.schema(), muts2);
                 underlying.apply(std::move(mt2));
```

Comment:
This code is clearly a test artifact and doesn't happen in the real code paths. So we aren't hiding a real-world problem.

The real code does have a similar exception safety problem though:

```c++
    auto adder = row_cache::external_updater([this, m, ssts = std::move(ssts), new_ssts_ms = std::move(*ms_opt), &cg] () mutable {
        // FIXME: the following isn't exception safe.
        for (auto& sst : ssts) {
            add_sstable(cg, sst);
            update_stats_for_new_sstable(sst);
        }
        m->mark_flushed(std::move(new_ssts_ms));
        try_trigger_compaction(cg);
    });
```

Reply:
@bhalevy you have a patch to fix this, right? can we extract it from the long series (IIRC) it belongs?

Reply:
It's https://github.com/scylladb/scylladb/pull/14043/commits/288a240e4eccb0aed128caa5d7aa1a62e4bc3dd1, part of https://github.com/scylladb/scylladb/pull/14043
but I'm not sure if it's worth trying to male it a standalone patch